src/extension.ts

```diff
@@ -206,36 +207,60 @@ export function activate(context: ExtensionContext) {
     });
 
     const engineDecoration = window.createTextEditorDecorationType({});
-    let latestDecs: any[] | undefined = undefined;
+    let latestDecorations: EngineDecoration[] | undefined = undefined;
 
     const updateDecorations = () => {
-      if (window.activeTextEditor && latestDecs) {
+      if (window.activeTextEditor && latestDecorations) {
         const editor = window.activeTextEditor!;
-        const decorations: DecorationOptions[] = latestDecs
+        const decorations = latestDecorations
           .filter(
-            (d) =>
-              d.document === window.activeTextEditor!.document.uri.toString()
+            (decoration) =>
+              decoration.document ===
+              window.activeTextEditor!.document.uri.toString()
           )
-          .map((dec) => {
+          .map((decoration): DecorationOptions => {
+            const onDiskPath = Uri.file(
+              join(context.extensionPath, "src", "iconRun.svg")
+            );
+
+            const hoverMessage =
+              decoration.hoverMessage === undefined
```

Comment:
nit: is this equivalent to `decoration.hoverMessage && new MarkdownString(decoration.hoverMessage)`

Reply:
The empty string being falsy makes me prefer explicit `undefined` checks 🤷‍♂️ I would rather have empty string show an empty hover.